app/src/main/java/org/dhis2/usescases/events/ScheduledEventActivity.kt

```diff
@@ -109,14 +104,17 @@ class ScheduledEventActivity : ActivityGlobalAbstract(), ScheduledEventContract.
                         dateValue = "",
                     )
 
+                    val eventDateAction = if (programStage.periodType() == null) null else showEventDatePeriodDialog(programStage.periodType())
```

Comment:
maybe here you can use an `programStage.periodType().?let.{
showEventDatePeriodDialog(it)
}`